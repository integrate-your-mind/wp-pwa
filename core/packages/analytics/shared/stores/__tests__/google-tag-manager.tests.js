import { unprotect } from 'mobx-state-tree';
import { Stores, itemCat7 } from '../mocks';

let stores;
beforeEach(() => {
  stores = Stores.create({});
  unprotect(stores);
});

describe('Analytics > GoogleTagManager', () => {
  test('sendPageView', () => {
    window.dataLayer = [];
    stores.analytics.googleTagManager.sendPageView();
    stores.connection.selectedItem = itemCat7;
    stores.analytics.googleTagManager.sendPageView();
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('sendPageView with title and location', () => {
    window.dataLayer = [];
    stores.analytics.googleTagManager.sendPageView({
      title: 'The Beauties of Gullfoss - page 2',
      location: 'https://demo.frontity.test/the-beauties-of-gullfoss/2',
    });
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('sendEvent', () => {
    const event = {
      category: 'test/category',
      action: 'test/action',
      label: 'test/label',
    };

    window.dataLayer = [];
    stores.analytics.googleTagManager.sendEvent(event);
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('view - clientProperties', () => {
    expect(
      stores.analytics.googleTagManager.clientProperties,
    ).toMatchSnapshot();
  });

  test('view - ampVars', () => {
    expect(stores.analytics.googleTagManager.ampVars).toMatchSnapshot();
  });

  test('view - ids', () => {
    stores.build = { dev: true, channel: 'pwa' };
    expect(stores.analytics.googleTagManager.ids).toMatchSnapshot();
    stores.build = { dev: true, channel: 'amp' };
    expect(stores.analytics.googleTagManager.ids).toMatchSnapshot();
  });
});
