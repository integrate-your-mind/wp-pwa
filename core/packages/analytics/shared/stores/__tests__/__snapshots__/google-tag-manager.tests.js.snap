// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics > GoogleTagManager sendEvent 1`] = `
Array [
  Object {
    "event": "virtualEvent",
    "virtualEvent": Object {
      "action": "PWA - test/action",
      "category": "PWA - test/category",
      "label": "test/label type: post context: single",
    },
  },
]
`;

exports[`Analytics > GoogleTagManager sendPageView 1`] = `
Array [
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": Object {
        "dimension1": "dim1_post60",
        "dimension2": "dim2_post60",
      },
      "format": "pwa",
      "hash": "ZO0H2I9kJ0arOdstZdG",
      "id": 60,
      "page": undefined,
      "route": "single",
      "site": "https://demo.frontity.test",
      "title": "The Beauties of Gullfoss",
      "type": "post",
      "url": "https://demo.frontity.test/the-beauties-of-gullfoss/",
    },
  },
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": Object {
        "dimension1": "dim1_cat7",
        "dimension2": "dim2_cat7",
      },
      "format": "pwa",
      "hash": "F1IyuIcMmLJVe2Je0OX",
      "id": 7,
      "page": 2,
      "route": "list",
      "site": "https://demo.frontity.test",
      "title": "Photography",
      "type": "category",
      "url": "https://demo.frontity.test/wp-cat/photography/page/2",
    },
  },
]
`;

exports[`Analytics > GoogleTagManager sendPageView with title and location 1`] = `
Array [
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": Object {
        "dimension1": "dim1_post60",
        "dimension2": "dim2_post60",
      },
      "format": "pwa",
      "hash": "ZO0H2I9kJ0arOdstZdG",
      "id": 60,
      "page": undefined,
      "route": "single",
      "site": "https://demo.frontity.test",
      "title": "The Beauties of Gullfoss - page 2",
      "type": "post",
      "url": "https://demo.frontity.test/the-beauties-of-gullfoss/2",
    },
  },
]
`;

exports[`Analytics > GoogleTagManager view - ampVars 1`] = `
Object {
  "anonymize": false,
  "customDimensions": Object {
    "dimension1": "dim1_post60",
    "dimension2": "dim2_post60",
  },
  "dev": true,
  "extensions": "saturn-theme,wp-org-connection",
  "format": "pwa",
  "hash": "ZO0H2I9kJ0arOdstZdG",
  "id": 60,
  "page": undefined,
  "pageType": "pwa",
  "plan": "enterprise",
  "route": "single",
  "site": "https://demo.frontity.test",
  "siteId": "site1122334455",
  "theme": "saturn-theme",
  "title": "The Beauties of Gullfoss",
  "type": "post",
  "url": "https://demo.frontity.test/the-beauties-of-gullfoss/",
  "userIds": Array [
    "user00",
    "user01",
  ],
}
`;

exports[`Analytics > GoogleTagManager view - clientProperties 1`] = `
Object {
  "anonymize": false,
  "dev": true,
  "extensions": "saturn-theme,wp-org-connection",
  "pageType": "pwa",
  "plan": "enterprise",
  "siteId": "site1122334455",
  "theme": "saturn-theme",
  "userIds": Array [
    "user00",
    "user01",
  ],
}
`;

exports[`Analytics > GoogleTagManager view - ids 1`] = `
Array [
  "GTM-112233",
]
`;

exports[`Analytics > GoogleTagManager view - ids 2`] = `
Array [
  "GTM-123456",
]
`;
